download/src/lib.rs

```diff
@@ -47,12 +47,14 @@ fn download_with_backend(
     }
 }
 
+type DownloadCallback<'a> = Option<&'a dyn Fn(Event<'_>) -> Result<()>>;
```

Comment:
This doesn't look right as a name to me. 

```
type DownloadCallback<'a> = &'a dyn Fn(Event<'_'>) -> Result<()>;
```
might, but will that make clippy happy? If not, I think we should push back on the lint.